optuna/integration/wandb.py

```diff
@@ -126,10 +147,43 @@ def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -
         metrics = {name: value for name, value in zip(names, trial.values)}
         attributes = {"direction": [d.name for d in study.directions]}
 
-        wandb.config.update(attributes)
-        wandb.log({**trial.params, **metrics}, step=trial.number)
+        step = trial.number if wandb.run else None
+        run = wandb.run
 
-    def _initialize_run(self) -> None:
-        """Initializes Weights & Biases run."""
+        # When an user doesn't use `track_in_wandb` decorator.
+        # Might create an extra run if an user logs in wandb but doesn't use the decorator.
```

Comment:
I'm afraid that I could not imagine this case. Could you share a minimal example code for this use case?

Reply:
Sure. For example an user can do something like this:

```python3
def objective(trial):
  with wandb.init() as run:
    run.log({"test": 0})
  return loss
```

Instead of doing the correct thing:

```python3

@track_in_wandb
def objective(trial):
  wandb.log({"test": 1})
```

If an user uses the first option, there's no way for us to reuse that run. Therefore, when we try to log the params, it would create another run to log it, instead of reusing the last one.

Reply:
Ah, i got it. Thank you for your explanaion.